closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -416,7 +475,7 @@ int decryptFinal(byte[] in, int inOfs, int len,
                      byte[] out, int outOfs)
         throws IllegalBlockSizeException, AEADBadTagException,
         ShortBufferException {
-        if (len < tagLenBytes) {
+        if (ibuffer.size() + len < tagLenBytes) {
```

Comment:
This should deal with integer overflow.

Reply:
If `len` and `tagLenBytes` have both been validates as non-negative, then Peter's comment was (I believe) that you should write this:
```
if (ibuffer.size() < tagLenBytes - len) {
```

Reply:
`ibuffer.size()` is added to `len` in multiple places. Would it make sense to make a new `long` variable to hold the sum?

Reply:
That doesn't address the potential for overflow (the conversion from `int` to `long` occurs after the addition).
`len` needs to be validated somewhere that it is non-negative, after which, `tagLenBytes - len` cannot overflow, hence my suggestion.

Reply:
The suggested solution would work when adding `ibuffer.size()` to `len` in this if statement. However, since I will be adding these two variables in 3 different places in that function, I am looking into a more generic solution. This is how I ended up solving it (in [commit 9cbcfc2](https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/266/commits/9cbcfc23e5e72be110c20579657695f469c466b4)):

I replace `len += ibuffer.size()` with:

```java
try {
    len = Math.addExact(len,ibuffer.size());
}
catch(ArithmeticException e) {
    throw new ProviderException("SunJCE provider only supports " +
        "input size up to " + MAX_BUF_SIZE + " bytes");
}
```
I throw an exception if an integer overflow happens because this is the intended behaviour. In `checkDataLength`, if `len + proc > Integer.MAX_VALUE`, the same exception is thrown. The check I add here is essentially the same check, but I add it earlier in the code.
Having said that, one might ask why do I keep `checkDataLength` if the same check is essentially happening earlier. This is because if we later decide to change the `MAX_BUF_SIZE` to a value smaller than `Integer.MAX_VALUE`, the check would not be redundant anymore. But for our current case, the second check does not add any value.